### ISSUE TITLE
CI: Required tests should be run in continuous integration

### ISSUE THREAD

**Issue by cclauss** [nodejs/nodejs.org#6264]
### Enter your suggestions in details:

When contributors make pull requests, they want to know if the continuous integration tests pass but four `required` tests do not run on every pull request.  Either these tests should __always__ be run or they should not be marked as `required`.

![Required tests should not be optional](https://github.com/nodejs/nodejs.org/assets/3709715/ef727fbc-62f3-4368-9691-6cd707925156)

**Comment by bmuenzenmeyer** [comment]
there's some security-related reasons we don't execute build and deploy on PRs - though I concede lint and test make a lot of sense. currently our guidance is that these commands should be run locally (communicated via the PR template, but that certainly doesn't address web-based contributions)

it's a balance of methodical improvement vs feedback, sometimes we get the developer experience wrong and it warrants conversations like this.

anything to add @ovflowd ?

**Comment by ovflowd** [comment]
All these steps are executed on a CI. Please read the Collaborator Guide. The reason why they aren't triggered by default is for security reasons.

**Comment by cclauss** [comment]
I understand the security reasons and how other repos have avoided them.  It is just frustrating for contributors to see there contributions permanently in a yellow state.

**Comment by ovflowd** [comment]
There are other reasons besides security. For example, regular "pull_request" events do not have permissions to allow comments to be added on a PR or to allow the Action itself to push a commit to the same repository. The "pull_request_target" permission allow us to have "write" permissions besides of "read" to numerous features of GitHub.

But "pull_request_target" bypasses the default "approve" the CI to run flow. Hence we made it to be behind a label.

I don't see how seeing that is frustrating. People already asked before why their CI is not running and we simply answered and then they were OK. CI shouldn't be a matter of frustration or of indication of success. They are just checks that need to be fulfilled before a PR can be merged.

For all practical reasons, it's even greener and resource friendly to only run the checks once we finished reviewing a PR and we believe it's ready to go. Instead of having the checks running in every commit, we decide when it is appropriate to run. Less resources wasted. Happier planet. Less noise. Less CI failures due to PRs being on draft or incomplete.
